File: Java/2024_cursoemvideo_poo/aula05_contaBanco/src/Conta.java
public class Conta {

    //Atributos

    private int num;
    private String nome;
    private String tipo;
    private float saldo;
    private boolean status;

    public Conta(int num, String nome, String tipo, float saldo, boolean status) {
        this.nome = nome;
        this.num = num;
        this.tipo = tipo;
        if (tipo == "Corrente") {
            this.saldo = saldo + 50;
        } else {
            this.saldo = saldo + 100;
        }
        this.status = status;
    }

    //GETS & SETS

    public int getNum() {
        return num;
    }

    public void setNum(int num) {
        this.num = num;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String isTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public float getSaldo() {
        return saldo;
    }

    public void setSaldo(float saldo) {
        this.saldo = saldo;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    //MÉTODOS

    public void Depositar(int x) {
        saldo += x;
        System.out.println("Depósito feito. Seu saldo é: " + saldo);
    }

    public void Sacar(int x) {
        if (x <= saldo) {
            saldo -= x;
            System.out.println("Saque realizado com sucesso.");
            System.out.println("Seu saldo é: " + saldo);
        } else {
            System.out.println("Não há saldo suficiente.");
        }
    }

    public void PagMensal() {
        if (tipo=="Corrente") {
            saldo -= 50;
        } else {
            saldo -= 100;
        }
    }
    public void FecharConta() {
        if (saldo == 0) {
            this.status = false;
            System.out.println("Conta fechada com sucesso!");
        } else {
            System.out.println("A conta precisa estar vazia para ser desativada.");
        }
    }

}

File: PHP/2016_cursoemvideo_poo/aula05_contaBancaria/Conta.php
<?php
class Conta {
    public int $num;
    protected ?string $tipo;
    private ?string $nome; 
    private int $saldo;
    private bool $status;

//    public function Conta($num, $nome, $tipo, $saldo) (outdated)
    public function __construct($num1, $nome, $tipo, $saldo) {
        $this->num = $num1;
        $this->nome = $nome;
        $this->tipo = $tipo;
        if ($tipo == "Corrente") {
            $this->saldo = $saldo + 50;
        } else {
            $this->saldo = $saldo + 100;
        }
        $this->status = true;
    }

    public function getNum() {
        return $this->num;
    }
    public function getNome() {
        return $this->nome;
    }
    public function getTipo() {
        return $this->tipo;
    }
    public function getSaldo() {
        return $this->saldo;
    }
    public function getStatus() {
        return $this->status;
    }
    public function Status() {
        echo $this->getSaldo();
    }


    //MÉTODOS
    
    public function Depositar($x) {
        $this->saldo = $this->saldo + $x;
        echo "Deposito de $x reais feito com sucesso. Seu saldo é $this->saldo.";
    }
    public function Sacar($x) {
        if ($x > $this->saldo) {
            echo "Saldo insuficiente. Seu saldo é $this->saldo.";
        } else {
            $this->saldo = $this->saldo - $x;
            echo "Saque de $x reais feito. Seu saldo é $this->saldo.";
        }
    }
    public function Pagar() {
        if ($this->saldo < 50) {
            echo "Saldo insuficiente. Seu saldo é $this->saldo.";
        } else {
            $this->saldo = $this->saldo - 50;
            echo "Mensalidade paga. Seu saldo é $this->saldo.";
        }
    }
    public function FecharConta() {
        if ($this->saldo != 0) {
            echo "Não é possível. A conta precisa estar zerada.";
        } else {
            $this->status = false;
            echo "Conta fechada com sucesso!";
        }
    }
}
?>


